experimental/devices/unicornhd/unicornhd_test.go

```diff
@@ -0,0 +1,247 @@
+package unicornhd
+
+import (
+	"bytes"
+	"errors"
+	"image"
+	"image/color"
+	"testing"
+
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/conn/spi/spitest"
+)
+
+func TestNewFailsWhenConnectionToSpiFails(t *testing.T) {
+	if dev, err := NewUnicornhd(&spiFail{}); dev != nil || err == nil {
+		t.Fatal()
+	}
+}
+
+func TestNewReturnsDriverWithGoodSpi(t *testing.T) {
+	if dev, err := NewUnicornhd(spitest.NewRecordRaw(nil)); dev == nil || err != nil {
+		t.Fatal()
+	}
+}
+
+func TestStringIsDriverNameWidthHeight(t *testing.T) {
+	expectedString := "UnicornHD{16, 16}"
+	dev, _ := NewUnicornhd(spitest.NewRecordRaw(nil))
+	devString := dev.String()
+	if devString != expectedString {
+		t.Fatalf("expected: '%s', actual: '%s'", expectedString, devString)
+	}
+}
+
+func TestColorModeIsNRGBA(t *testing.T) {
+	dev, _ := NewUnicornhd(spitest.NewRecordRaw(nil))
+
+	if dev.ColorModel() != color.NRGBAModel {
+		t.Fatal()
+	}
+}
+
+func TestBoundsMatchDeviceSize(t *testing.T) {
+	dev, _ := NewUnicornhd(spitest.NewRecordRaw(nil))
+
+	bounds := dev.Bounds()
+
+	if bounds.Min.X != 0 {
+		t.Errorf("Min x expected to be zero (actual: %d)", bounds.Min.X)
+	}
+	if bounds.Min.Y != 0 {
+		t.Errorf("Min y expected to be zero (actual: %d)", bounds.Min.Y)
+	}
+	if bounds.Max.X != Width {
+		t.Errorf("Max x expected to be %d (actual: %d)", Width, bounds.Max.X)
+	}
+	if bounds.Max.Y != Height {
+		t.Errorf("Max y expected to be %d (actual: %d)", Height, bounds.Max.Y)
+	}
+}
+
+func TestDrawWritesBlackImageToSpi(t *testing.T) {
+	var expectedAllBlack = []byte{
+		0x72,
+		0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
```

Comment:
Since it's all zeros, you could do:

    var expectedAllBlack = append([]byte{0x72}, make([]byte, 16*16*3)...)